### DISCUSSION TITLE
Europas Finest Achievement

### DISCUSSION THREAD

**Discussion by Redtailxmas** [FakeFishGames/Barotrauma#15444] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

The steam achievement "Europa's Finest" did not unlock after my group completed the campaign on Ironman mode. I'm also not sure if there's just more of an unspoken criteria for the achievement such as completing a second campaign on the same save. Or perhaps starting a fresh campaign and leaving the server running in the background until the campaign is complete. I figured I'd ask before committing time to something that could be potentially bugged.

### Reproduction steps

1. Completed a multiplayer campaign on Ironman mode
2. Checked steams response to completion
3. Found "Europa's Finest" to still be a locked achievement 

### Bug prevalence

Just once

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.7.7.0 (Winter Update)

### Which operating system did you encounter this bug on?

Windows

**Comment by NotWendy** [comment]
I've seen people get that achievement without meeting any special, unspoken requirements. They just started the campaign with Ironman mode enabled and made sure it stayed on every time they loaded the save file, until they eventually got to the end.

**Comment by zMayTricks** [comment]
Also had this happen for me. Had permadeath and ironman mode both on. No mods. No cheats. Only 1 bot died. My character never died. I think the bug happens when resuming a campaign, because I noticed the permadeath Cold Caverns achievement only unlocked when I did the entire biome in one sitting. I'm not planning to redo an entire campaign on ironman mode in one sitting though, so I can't confirm it.

**Comment by Badstormer** [comment]
I've also had this issue after completing a campaign on permadeath mode without ironman enabled with the achievement not unlocking as expected (the achievement specifies permadeath **or** ironman mode) with no player or bot deaths. Loading a backup save before the final boss and double-checking that permadeath was enabled also did not unlock the achievement. Loading a backup again and ensuring permadeath was active _and_ toggling on ironman before loading successfully unlocked the achievement.

I suspect the save file silently reverts to default settings when loading the save without warning the host, leading a permadeath save to revert to a mid-round respawn save without the host realizing it. It's also possible the achievement isn't properly wired to accept permadeath and ironman independent of each other, given the only successful unlock happened specifically with ironman enabled.

I recall seeing commentary elsewhere that one's save file had the permadeath setting toggled off at some point without anyone realizing it, so settings silently getting changed when loading save files strikes me as the most likely culprit.

**edit:** after further testing, the save file is almost certainly being changed back to mid-round respawn when reloading without warning the player, as the default save file settings get applied when booting up a server and overwrite the existing settings if they aren't changed back every single time one loads a save. It does not, however, seem to toggle off ironman mode (so it's truly confusing why some people would not have the achievement unlocked despite running the campaign on ironman). The game **does not warn the host** when the respawn settings get overwritten by the default settings when loading a save.

**edit 2:** since ironman seems to only apply when permadeath is active, if the save file silently reverts to non-permadeath, the ironman toggle may be made redundant, leading to it also unexpectedly no longer being in effect
